astropy/units/tests/test_units.py

```diff
@@ -983,3 +983,23 @@ def test_hash_represents_unit(unit, power):
     assert hash(tu) == hash(unit)
     tu2 = (unit ** (1 / power)) ** power
     assert hash(tu2) == hash(unit)
+
+
+def test_dask_arrays():
```

Comment:
Should we use `from astropy.utils.compat.optional_deps import HAS_DASK` and then `pytest.skipif(not HAS_DASK, reason="tests dask.array")` here, for compatibility with other code?